packages/react/fast-components-react-base/src/data-grid/data-grid.tsx

```diff
@@ -0,0 +1,1103 @@
+                        this.props.gridData.length - 1
+                    );
+                    focusRowKey = this.props.gridData[focusRowIndex][
+                        this.props.dataRowKey
+                    ];
+                } else {
+                    focusRowIndex = newFocusRowIndex;
+                }
+
+                // ensure data page conforms to new data length
+                currentDataPageStartIndex =
+                    focusRowIndex - Math.floor(this.props.pageSize / 2);
+
+                currentDataPageStartIndex =
+                    currentDataPageStartIndex < 0 ? 0 : currentDataPageStartIndex;
+
+                currentDataPageEndIndex = currentDataPageStartIndex + this.props.pageSize;
+
+                currentDataPageEndIndex =
+                    currentDataPageEndIndex > this.props.gridData.length - 1
+                        ? this.props.gridData.length - 1
+                        : currentDataPageEndIndex;
+
+                this.sizeRowsToIndex(currentDataPageEndIndex, rowPositions);
+                estimatedTotalHeight = this.getEstimatedTotalHeight(rowPositions);
+
+                // move focus to the new element if necessary
+                if (
+                    (this.isFocused && this.state.focusRowKey !== focusRowKey) ||
+                    this.state.focusColumnKey !== focusColumnKey
+                ) {
+                    const rowElement: Element = this.getRowElementByKey(focusRowKey);
+                    if (rowElement !== null) {
+                        // next focus element already exists, focus on it
+                        (rowElement as HTMLElement).focus();
+                        desiredFocusRowKey = null;
+                        desiredFocusColumnKey = null;
+                    } else {
+                        // next focus element not in the dom,
+                        // focus on it after when it mounts
+                        desiredFocusRowKey = focusRowKey;
+                        desiredFocusColumnKey = focusColumnKey;
+
+                        // let the default behavior scroll into view on focus
+                        desiredVisibleRowIndex = null;
+                    }
+                }
+            }
+        }
+
+        if (shouldUpdateState) {
+            this.setState({
+                rowPositions,
+                currentDataPageStartIndex,
+                currentDataPageEndIndex,
+                estimatedTotalHeight,
+                focusRowKey,
+                focusColumnKey,
+                focusRowIndex,
+                desiredFocusRowKey,
+                desiredFocusColumnKey,
+                desiredVisibleRowIndex,
+            });
+        }
+    }
+
+    /**
+     * Generates class names
+     */
+    protected generateClassNames(): string {
+        const { dataGrid }: DataGridClassNameContract = this.props.managedClasses;
+
+        return super.generateClassNames(classNames(dataGrid));
```

Comment:
don't need to use classNames in this instance since you only ever return dataGrid